templates/cloud-init.sh

```diff
@@ -1,14 +1,40 @@
 #!/bin/bash
 
 set -x
-exec &> /tmp/cloud-init.log
 
 %{ for config_key, config_value in proxy_config ~}
 %{ if config_value != null ~}
 export ${config_key}="${config_value}"
 %{ endif ~}
 %{ endfor ~}
 
+# Checking if HTTP Proxy(s) provided and setting
+if [[ -z ${http_proxy} ]]; then
+  echo "No HTTP Proxy configuration set"
+else
+  echo "http_proxy=${http_proxy}" >> /etc/environment
+  touch /etc/apt/apt.conf.d/proxy.conf
+  echo "Acquire::http::Proxy ${http_proxy};" >> /etc/apt/apt.conf.d/proxy.conf
+fi
```

Comment:
you could do this with Terraform template variable checking, like this:
```suggestion
# Checking if HTTP Proxy(s) provided and setting
%{ if proxy_config.http_proxy != null ~}
  echo "http_proxy=${http_proxy}" >> /etc/environment
  touch /etc/apt/apt.conf.d/proxy.conf
  echo "Acquire::http::Proxy ${http_proxy};" >> /etc/apt/apt.conf.d/proxy.conf
%{ endif ~}
```

Reply:
Updated, pipeline now running green :D